storage_service/locations/models/dspace_rest.py

```diff
@@ -0,0 +1,494 @@
+import requests
+from requests import RequestException
+
+# This project, alphabetical
+from agentarchives import archivesspace
+from agentarchives.archivesspace.client import CommunicationError
+
+# This module, alphabetical
+from common import utils
+from .location import Location
+
+LOGGER = logging.getLogger(__name__)
+
+
+AS_DO_ADD_ERR = 'Could not add digital object to ArchivesSpace'
+
+
+class DSpaceRESTException(Exception):
+    pass
+
+
+class DSpaceREST(models.Model):
+    """Integration with DSpace using the REST API."""
+    space = models.OneToOneField('Space', to_field='uuid')
+    ds_rest_url = models.URLField(
+        max_length=256,
+        verbose_name=_l("REST URL"),
+        help_text=_l('URL of the REST API. E.g. http://demo.dspace.org/rest'))
+
+    ds_user = models.CharField(
+        max_length=64,
+        verbose_name=_l("User"),
+        help_text=_l('DSpace username to authenticate as'))
+
+    ds_password = models.CharField(
+        max_length=64,
+        verbose_name=_l("Password"),
+        help_text=_l('DSpace password to authenticate with'))
+
+    ds_dip_collection = models.CharField(
+        max_length=64,
+        verbose_name=_l("Default DSpace DIP collection id"),
+        help_text=_l('UUID of default DSpace collection for the DIP to be'
+                     ' deposited to'))
+
+    ds_aip_collection = models.CharField(
+        max_length=64,
+        verbose_name=_l("Default DSpace AIP collection id"),
+        help_text=_l('UUID of default DSpace collection for the AIP to be'
+                     ' deposited to'))
+
+    as_url = models.URLField(
+        blank=True,
+        max_length=256,
+        verbose_name=_l("ArchivesSpace URL"),
+        help_text=_l('URL of ArchivesSpace server. E.g.'
+                     ' http://sandbox.archivesspace.org:8089/ (default port'
+                     ' 8089 if omitted)'))
+
+    as_user = models.CharField(
+        blank=True,
+        max_length=64,
+        verbose_name=_l("ArchivesSpace user"),
+        help_text=_l('ArchivesSpace username to authenticate as'))
+
+    as_password = models.CharField(
+        blank=True,
+        max_length=64,
+        verbose_name=_l("ArchivesSpace password"),
+        help_text=_l('ArchivesSpace password to authenticate with'))
+
+    as_repository = models.CharField(
+        blank=True,
+        max_length=64,
+        verbose_name=_l("Default ArchivesSpace repository"),
+        help_text=_l('Identifier of the default ArchivesSpace repository'))
+
+    as_archival_object = models.CharField(
+        blank=True,
+        max_length=64,
+        verbose_name=_l("Default ArchivesSpace archival object"),
+        help_text=_l('Identifier of the default ArchivesSpace archival object'))
+
+    verify_ssl = models.BooleanField(
+        blank=True, default=True, verbose_name=_l("Verify SSL certificates?"),
+        help_text=_l('If checked, HTTPS requests will verify the SSL'
+                     ' certificates'))
```

Comment:
Thanks @sevein, I had forgotten about `INSECURE_SKIP_VERIFY`. I think it's good to keep `DSpaceREST.verify_ssl` because, as you say, it is more fine-grained. This allows users to decrease their security wrt trusting responses from external services in the specific DSpace REST case but not elsewhere.